airbyte-cdk/bulk/toolkits/load-low-code/src/main/kotlin/io/airbyte/cdk/load/lowcode/DeclarativeDestinationFactory.kt

```diff
@@ -199,4 +251,103 @@ class DeclarativeDestinationFactory(config: JsonNode?) {
 
     private fun createInterpolationContext(): Map<String, Any> =
         mapOf("config" to Jsons.convertValue(config, MutableMap::class.java))
+
+    fun DynamicCatalogOperationModel.toProvider(): DynamicOperationProvider {
+        val dynamicOperation = this
+        return DynamicOperationProvider(
+            createDestinationObjectProvider(dynamicOperation.objects),
+            createDestinationOperationAssembler(dynamicOperation)
+        )
+    }
+
+    private fun createDestinationObjectProvider(
+        model: DestinationObjectsModel
+    ): DestinationObjectProvider =
+        when (model) {
+            is StaticDestinationObjectsModel -> StaticDestinationObjectProvider(model.objects)
+            is DynamicDestinationObjectsModel ->
+                DynamicDestinationObjectProvider(
+                    model.retriever.toRetriever(),
+                    model.namePath ?: emptyList()
+                )
+        }
+
+    private fun createDestinationOperationAssembler(
+        model: DynamicCatalogOperationModel
+    ): DestinationOperationAssembler {
+        val propertiesPath: List<String> = model.schema.propertiesPath
+        val insertionMethods: List<InsertionMethod> =
+            model.insertionMethods.map { createInsertionMethod(model.schema, it) }
+        val schemaRequester: HttpRequester? = model.schemaRetriever?.httpRequester?.toRequester()
+        return DestinationOperationAssembler(propertiesPath, insertionMethods, schemaRequester)
+    }
+
+    fun RetrieverModel.toRetriever(): Retriever {
+        val retriever = this
+        val requester: HttpRequester = retriever.httpRequester.toRequester()
+        return Retriever(requester, retriever.selector)
+    }
+
+    private fun createInsertionMethod(
+        schemaModel: SchemaConfigurationModel,
+        model: InsertionMethodModel
+    ): InsertionMethod =
+        InsertionMethod(
+            mapImportMode(model.destinationImportMode),
+            schemaModel.propertyNamePath,
+            schemaModel.typePath,
+            JsonNodePredicate(model.availabilityPredicate),
+            model.matchingKeyPredicate?.let { JsonNodePredicate(model.matchingKeyPredicate) },
+            JsonNodePredicate(model.requiredPredicate),
+            schemaModel.typeMapping?.let { createTypeMapping(it) } ?: emptyMap(),
+        )
+
+    private fun createTypeMapping(model: List<TypesMapModel>): Map<String, AirbyteType> =
+        /*
+         *  mapping.airbyteType[0] is not what we want long term. The protocol allows for an API
+         *  type to map to multiple types, but this doesn't map to the runtime InsertionMethod's
+         *  typeMapping which is 1:1 between API type -> Airbyte type. If multiple are provided
```

Comment:
Should we fail loudly instead? I fear a connector developer might be surprised by this behavior and lose some time